src/hooks/usePatchStatusSelect.ts

```diff
@@ -1,22 +1,77 @@
-import { useState, useEffect } from "react";
+import { useEffect, useReducer, useState } from "react";
 import { PatchBuildVariant } from "gql/generated/types";
+import { usePrevious } from "hooks";
+import { getWebWorkerURL } from "utils/getEnvironmentVariables";
 
 export interface selectedStrings {
   [id: string]: boolean | undefined;
 }
 
-export const usePatchStatusSelect = (
-  patchBuildVariants: PatchBuildVariant[]
-): [
+type Action =
+  | { type: "setSelectedTasks"; data: selectedStrings }
+  | { type: "setPatchStatusFilterTerm"; data: string[] }
+  | { type: "setBaseStatusFilterTerm"; data: string[] };
+
+interface State {
+  patchStatusFilterTerm: string[];
+  baseStatusFilterTerm: string[];
+  selectedTasks: selectedStrings;
+}
+
+const reducer = (state: State, action: Action) => {
+  switch (action.type) {
+    case "setSelectedTasks":
+      return {
+        ...state,
+        selectedTasks: action.data,
+      };
+    case "setPatchStatusFilterTerm":
+      return {
+        ...state,
+        patchStatusFilterTerm: action.data,
+      };
+    case "setBaseStatusFilterTerm":
+      return {
+        ...state,
+        baseStatusFilterTerm: action.data,
+      };
+    default:
+      throw new Error();
+  }
+};
+
+type FilterSetter = (statuses: string[]) => void;
+
+type HookResult = [
   selectedStrings,
   string[],
+  string[],
   {
-    setValidStatus: (newValue: string[]) => void;
     toggleSelectedTask: (id: string | string[]) => void;
+    setPatchStatusFilterTerm: FilterSetter;
+    setBaseStatusFilterTerm: FilterSetter;
   }
-] => {
-  const [selectedTasks, setSelectedTasks] = useState<selectedStrings>({});
-  const [validStatus, setValidStatus] = useState<string[]>([]);
+];
+
+export const usePatchStatusSelect = (
+  patchBuildVariants: PatchBuildVariant[]
+): HookResult => {
+  const [webWorker, setWebWorker] = useState<Worker>();
+  useEffect(() => {
+    if (window.Worker && !webWorker) {
+      const worker = new Worker(getWebWorkerURL("patchBuildVariantsReduce.js"));
+      setWebWorker(worker);
+    }
+  }, [webWorker]);
```

Comment:
Neat! i think this logic should be extracted to its own hook if possible. Especially if you think this is something we might do more of.